sqlmesh/core/context.py

```diff
@@ -800,16 +841,14 @@ def plan(
         model_selector = Selector(
             self.state_reader,
             self._models,
-            {
-                model.name: self.config_for_node(model).model_defaults
-                for model in self.models.values()
-            },
-            self.path,
+            context_path=self.path,
+            default_catalog=self.default_catalog,
+            dialect=self.config.dialect,
         )
 
-        models_override: t.Optional[UniqueKeyDict[str, Model]] = None
+        model_override: UniqueKeyDict[str, Model] = UniqueKeyDict("models")
```

Comment:
I'd prefer for this to remain optional rather than rely on an empty dict. Empty dict may mean no models rather than no overrides.